sdk/src/main/java/com/google/cloud/dataflow/sdk/util/common/Counter.java

```diff
@@ -241,6 +246,35 @@ public AggregationKind getKind() {
    */
   public abstract T getAggregate();
 
+  /** Returns the naming context associated with this counter. */
+  public NameContext getNameContext() {
+    return this.name.getContext();
+  }
+
+  /** Sets the naming context associated with this counter. */
+  public void setNameContext(NameContext context, NameType type) {
+    if (context != null) {
+      switch (type) {
+          case UNSTRUCTURED:
+            throw new IllegalArgumentException("Unstructured counters have no context");
```

Comment:
I wonder if it would be better to have an "UnstructuredName(this.name.getName())" that we use here?